src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -168,2164 +193,2043 @@ void *frontendservicesesion(std::string masterIP, int connFd, SQLiteDBInterface
+            vertex_count_command(connFd, sqlite, &loop_exit);
+        } else if (line.compare(ECOUNT) == 0) {
+            edge_count_command(connFd, sqlite, &loop_exit);
+        } else if (line.compare(MERGE) == 0) {
+            merge_command(connFd, sqlite, &loop_exit);
+        } else if (line.compare(TRAIN) == 0) {
+            train_command(connFd, sqlite, &loop_exit);
+        } else if (line.compare(IN_DEGREE) == 0) {
+            in_degree_command(connFd, &loop_exit);
+        } else if (line.compare(OUT_DEGREE) == 0) {
+            out_degree_command(connFd, &loop_exit);
+        } else if (line.compare(PAGE_RANK) == 0) {
+            page_rank_command(connFd, &loop_exit);
+        } else if (line.compare(EGONET) == 0) {
+            egonet_command(connFd, &loop_exit);
+        } else if (line.compare(DPCNTRL) == 0) {
+            duplicate_centralstore_command(connFd, &loop_exit);
+        } else if (line.compare(PREDICT) == 0) {
+            predict_command(masterIP, connFd, sqlite, &loop_exit);
+        } else if (line.compare(START_REMOTE_WORKER) == 0) {
+            start_remote_worker_command(connFd, &loop_exit);
+        } else if (line.compare(SLA) == 0) {
+            sla_command(connFd, sqlite, perfSqlite, &loop_exit);
+        } else {
+            frontend_logger.error("Message format not recognized " + line);
+            // TODO: Inform client?
+        }
+    }
+    if (input_stream_handler.joinable()) {
+        input_stream_handler.join();
+    }
+    frontend_logger.info("Closing thread " + to_string(pthread_self()) + " and connection");
+    close(connFd);
+    return NULL;
+}
 
-            // We get the name and the path to graph as a pair separated by |.
-            char graph_data[FRONTEND_DATA_LENGTH + 1];
-            bzero(graph_data, FRONTEND_DATA_LENGTH + 1);
-            string name = "";
-            string path = "";
+JasmineGraphFrontEnd::JasmineGraphFrontEnd(SQLiteDBInterface db, PerformanceSQLiteDBInterface perfDb,
+                                           std::string masterIP, JobScheduler jobScheduler) {
+    this->sqlite = db;
+    this->masterIP = masterIP;
+    this->perfSqlite = perfDb;
+    this->jobScheduler = jobScheduler;
+}
 
-            read(connFd, graph_data, FRONTEND_DATA_LENGTH);
+int JasmineGraphFrontEnd::run() {
+    int pId;
+    int portNo = Conts::JASMINEGRAPH_FRONTEND_PORT;
+    int listenFd;
+    socklen_t len;
+    bool loop = false;
+    struct sockaddr_in svrAdd;
+    struct sockaddr_in clntAdd;
 
-            std::time_t time = chrono::system_clock::to_time_t(chrono::system_clock::now());
-            string uploadStartTime = ctime(&time);
-            string gData(graph_data);
+    // create socket
+    listenFd = socket(AF_INET, SOCK_STREAM, 0);
 
-            gData = Utils::trim_copy(gData, " \f\n\r\t\v");
-            frontend_logger.log("Data received: " + gData, "info");
+    if (listenFd < 0) {
+        frontend_logger.error("Cannot open socket");
```

Comment:
Shall we add bit more details here like the port number on which the socket opening got failed?

Reply:
The port or host has no effect here. It is the socket creation that has failed. Port and host has an effect only when and after binding, which happens later.